### DISCUSSION TITLE
How to define chunk names in 3rd party code?

### DISCUSSION THREAD

**Discussion by scameron** [webpack/webpack#18731] [General]
Sometimes we encounter 3rd party code that contains async imports, which will cause webpack to create a new chunk group.  If there is no `webpackChunkName` magic comment defined, the chunk group will be anonymous and the final chunk name will be generic.  Furthermore, each async import creates a separate chunk group, leading (potentially) to many small network requests.

I believe the only way to control the chunk group definitions is via the `webpackChunkName` magic comments.  But what is the recommendation for 3rd part code?  Is there a good way to inject magic comments into such code at build time?

I found this open source project that uses a loader:
https://github.com/morganney/magic-comments-loader

That's an option but it looks like it parses each module again so I was hoping there might be a more "built-in" alternative approach that leverages the parsing already done by webpack itself.

Any ideas?

**Comment by alexander-akait** [comment]
We can provide an option for this, but yes, https://github.com/morganney/magic-comments-loader should work fine
